test/base-test.cc

```diff
@@ -425,6 +425,21 @@ TEST(arg_test, pointer_arg) {
   CHECK_ARG_SIMPLE(cp);
 }
 
+TEST(arg_test, volatile_pointer_arg) {
+  auto check = [](const int val) {
+    CHECK_ARG(
+        char, reinterpret_cast<const void*>(val),
+        static_cast<volatile void*>(reinterpret_cast<volatile int*>(val)));
+    CHECK_ARG(char, reinterpret_cast<const void*>(val),
+              static_cast<const volatile void*>(
+                  reinterpret_cast<const volatile int*>(val)));
+  };
+  check(0);
+  check(1);
+  check(2);
+  check(0xbeef);
```

Comment:
I don't think we need to check various values here, let's get rid of the lambda and just check nullptr.

Reply:
I was just being safe to cover the behaviour with iostreams linked in https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2021/p1147r1.html but if you're happy for it to be removed I'll take it out